Dockerfile

```diff
@@ -0,0 +1,58 @@
+ARG BASE_IMAGE=base
+FROM ros:jazzy AS base
+
+# Edit apt config for caching and update once
+RUN mv /etc/apt/apt.conf.d/docker-clean /etc/apt/ && \
+    echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' \
+      > /etc/apt/apt.conf.d/keep-cache && \
+    apt-get update
+
+################################################################################
+# MARK: prepper - prep rgl dependencies
+################################################################################
+### Core dependencies stage
+FROM $BASE_IMAGE AS prepper
+
+# Set working directory using standard opt path
+WORKDIR /opt/rgl
+
+# Copy package manifest
+COPY package.xml src/RGLGazeboPlugin/package.xml
+
+# Install bootstrap tools for install scripts
+RUN --mount=type=cache,sharing=locked,target=/var/cache/apt \
+    rosdep install -y \
+        --from-paths src \
+        --ignore-src
+
+################################################################################
+# MARK: builder - build rgl binaries
+################################################################################
+FROM prepper AS builder
+
+# Copy source tree
+COPY . src/RGLGazeboPlugin
+
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+    colcon build
+
+################################################################################
+# MARK: dancer - multi-stage for cache dancing
+################################################################################
+FROM builder AS dancer
+
+# Copy entire build directory
+# RUN mkdir /dancer && \
+#     cp -rT build /dancer
+
+# Copy only the lib and bin directories
+RUN mkdir /dancer && \
+    cp build/RGLGazeboPlugin/RGLServerPlugin/*.so /dancer/ && \
```

Comment:
From some reason, it doesn't seem like the `libRobotecGPULidar.so` is being properly installed into the colcon install folder, let alone the build folder.

- https://github.com/RobotecAI/RGLGazeboPlugin/blob/01d26ebf63507aa8ae9c52f6a226eece3f4409a8/RGLServerPlugin/CMakeLists.txt#L73-L75

@msz-rai do you know why this CMake project is writing to the src folder? That should be considered a read only directory.

Reply:
> do you know why this CMake project is writing to the src folder? That should be considered a read only directory.

Good point. At the very early development stage of this Plugin, the Library files were downloaded into src and that's how it stayed. This is not a great practice. I have separated it in [this PR](https://github.com/RobotecAI/RGLGazeboPlugin/pull/42). You are welcome to review it at your convenience.

> From some reason, it doesn't seem like the libRobotecGPULidar.so is being properly installed into the colcon install folder, let alone the build folder.

Could you expand on your issue? Do you mean the way we are installing is incorrect or the library is not present in the directory you expect it to be? I have made a quick test running `colcon build` on my PC and the library was installed into `{PluginDir}/install/RGLServerPlugin/libRobotecGPULidar.so` successfully.

Reply:
> Could you expand on your issue?

Please see https://github.com/RobotecAI/RGLGazeboPlugin/pull/40#issuecomment-2226038218 as a concrete test case.

Reply:
@ruffsl
Thank you for the explanation.

I have added to [my PR](https://github.com/RobotecAI/RGLGazeboPlugin/pull/42) two commits that:
- Remove `CMAKE_INSTALL_PREFIX` hardcoding
- Change the download location of `RobotecGPULidar` project from the source directory to the binary directory.

These modifications required changing the way how to use the custom build of RobotecGPULidar. [This section of the README](https://github.com/RobotecAI/RGLGazeboPlugin/tree/fix/separate-external-project?tab=readme-ov-file#using-custom-build-of-robotecgpulidar) may be useful for your integration.

Please test if my proposed changes resolve your issue. I have tested it (rebased your branch onto my branch) and it worked well for me. Note: One change is required after rebasing - `!/external` needs to be added to the `.dockerignore` file.

Reply:
Awesome @msz-rai , thanks! I tried by merging your PR with the above changes. Looks to be working, so I've simplified the Dockerfile to reflect that.